### DISCUSSION TITLE
Support styled scrollbars in Firefox

### DISCUSSION THREAD

**Discussion by PeterCxy** [standardnotes/forum#2610] [Suggestions]
Currently the scrollbars look pretty ugly in Firefox because the custom scrollbar styles uses webkit-specific extensions. However, Firefox has had support for standardized scrollbar styling for quite a while. Though I don't know if it's helpful, but here's another project that has add support for custom scrollbar styling in Firefox: https://github.com/tootsuite/mastodon/pull/8653 https://github.com/tootsuite/mastodon/pull/9210.

**Comment by atmoio** [comment]
Hmm interesting. Do you have comparison screenshots by any chance? I'm assuming this is on Windows right?

cc @baptiste-grob

**Comment by arielsvg** [comment]
No need for a screenshot, I can see it uses the default windows scrollbars.

**Comment by arielsvg** [comment]
Thanks for including references to prior art! Turns out aside from setting the color and width of the scrollbar, we can't do anything else (Per the CSS scrollbar module 1, which is what Firefox supports.) A thinner scrollbar _does_ look nicer… But it then becomes impossible to interact with because it overlaps with our panel resizers 😟 We'll have to think about this from a usability approach as well as an aesthetic one…

**Comment by PeterCxy** [comment]
@baptiste-grob I reckon just setting the color is enough to make it at least look okay. The current look is really irritating especially when a non-default theme is applied.

**Comment by arielsvg** [comment]
Can you show me a screenshot? Depending on your operating system it might look different from what I'm picturing

**Comment by PeterCxy** [comment]
![2020-02-26_19-42](https://user-images.githubusercontent.com/4532423/75341791-2840d200-58d0-11ea-8fb0-beae24e730fa.png)

This is how it looks on Linux with KDE

**Comment by arielsvg** [comment]
Ooooh I see what you mean now! It probably would look okay in the default theme, which is what I had in mind 😅 Yes, we'll definitely have to update every theme to support these new scrollbar properties.

**Comment by pwseo** [comment]
A year has passed since the last reply to this issue. Can't SN at least ship with a generic `scrollbar-color: var(...) var(...);` property referencing the theme's colors?
The screenshot posted above is from a light theme, but in dark themes (any of them) it looks absolutely awful, because it contrasts way too much, making them truly unusable:
![image](https://user-images.githubusercontent.com/95953/121449330-80023000-c991-11eb-86a7-56032696a60a.png)
Yes, the ugly bars in the image are from the scrollbar.

**Comment by JaspalSuri** [comment]
Hi @pwseo, thanks for the suggestion; I'll forward it over to the dev team for consideration. 🙂
